components/gitpod-db/src/typeorm/migration/1664799675276-MakeUidColumnPrimaryKeyInAuthCodeTable.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+const TABLE_NAME = "d_b_oauth_auth_code_entry";
+const PK_COLUMN = "id";
+const UID_COLUMN = "uid";
+
+export class MakeUidColumnPrimaryKeyInAuthCodeTable1664799675276 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`UPDATE d_b_oauth_auth_code_entry SET uid=(SELECT uuid()) where uid='';`);
+
+        await queryRunner.query(`ALTER TABLE ${TABLE_NAME} DROP COLUMN ${PK_COLUMN}`);
```

Comment:
Hm, is this really possible? :thinking: Have to look it up, actually.

Did you try to run this against an existing DB in the previous state?

Reply:
Looks like you can do this in one `ALTER TABLE` statement: https://stackoverflow.com/a/2342193

Reply:
> In general it appears you can drop PRIMARY KEY columns without issue.

Just verified: that's the case indeed. :+1: 

Also, because we run this inside a transaction, the `ADD PRIMARY KEY` should work, because we fill `uid` with concrete values before the move. :heavy_check_mark:.